ModelBase/src/nodes/Node.cpp

```diff
@@ -144,6 +144,13 @@ bool Node::isAncestorOf(const Node* other) const
 	return p == this;
 }
 
+Node* Node::firstAncestorOfType(const QString& typeName) const
```

Comment:
I would generalize this function and use a Matcher instead of a QString. It will be equally fast in case the matcher is a direct string comparison, but will also allow the extra flexibility.